bundles/org.eclipse.passage.lic.base/src/org/eclipse/passage/lic/base/conditions/LicensingConditions.java

```diff
@@ -62,28 +65,31 @@ public static LicensingCondition create(String identifier, String version, Strin
 	}
 
 	public static LicensingResult validate(LicensingCondition condition, String source) {
+		List<LicensingResult> details = new ArrayList<>();
+		Map<String, Object> data = new HashMap<String, Object>();
+		data.put(LicensingCondition.class.getName(), condition);
 		Date validFrom = condition.getValidFrom();
 		if (validFrom == null) {
 			String format = BaseMessages.getString("LicensingConditions_validation_no_from"); //$NON-NLS-1$
 			String message = String.format(format, condition);
-			return LicensingResults.createError(message, source);
+			return new BaseLicensingResult(LicensingResult.ERROR, message, 400, source, null, details, data);
 		}
 		Date now = new Date();
 		if (validFrom.after(now)) {
 			String format = BaseMessages.getString("LicensingConditions_validation_invalid_from"); //$NON-NLS-1$
 			String message = String.format(format, condition);
-			return LicensingResults.createError(message, source);
+			return new BaseLicensingResult(LicensingResult.ERROR, message, 416, source, null, details, data);
 		}
 		Date validUntil = condition.getValidUntil();
 		if (validUntil == null) {
 			String format = BaseMessages.getString("LicensingConditions_validation_no_until"); //$NON-NLS-1$
 			String message = String.format(format, condition);
-			return LicensingResults.createError(message, source);
+			return new BaseLicensingResult(LicensingResult.ERROR, message, 400, source, null, details, data);
```

Comment:
Here reason code is equal to the case of `null` in `validFrom`. Looks suspicious.

Reply:
Yes, both cases are kinds of "invalid request"